### PR TITLE
deluge: 1.3.15 -> 2.0.3 [WIP]

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Upgrade from 1.3.15 to 2.0.3. At the same time we move to py3 and gtk3.

I know that @FRidh is hesitant to add new packages but the problem here is that we don't carry ```pytest-twisted``` which is needed for the tests to run.

Also a ```libtorrent-rasterbar``` update to go along with this.

The toolbar icons don't look great here, but that may be due to my theme/settings.

WIP because the libtorrent change breaks qbittorrent.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @Anton-Latukha @Phreedom @domenkozar @ebzzry

### CODE REVIEW

**Review by Anton-Latukha — APPROVED** [review]
I observed the `libtorrentRasterbar` changes.

**Comment by Infinisil** [comment]
There's already https://github.com/NixOS/nixpkgs/pull/64542

File: pkgs/applications/networking/p2p/deluge/default.nix [hunk 1]
```diff
@@ -1,39 +1,47 @@
...
+  postPatch = ''
+    echo ${version} > RELEASE-VERSION
+  '';
+
+  buildInputs = lib.optionals withGui [ gobject-introspection gtk3 libnotify librsvg ];
+
+  propagatedBuildInputs = with python3Packages; [
+    dbus-python geoip2 pillow rencode setproctitle six
+    twisted Mako chardet pyxdg pyopenssl service-identity
+    (libtorrentRasterbar.override { inherit (python3Packages) python; }).python
```
Comment (FRidh) [inline pkgs/applications/networking/p2p/deluge/default.nix]: Why is this done. `libtorrentRasterbar` exists in `python-packages.nix` and should be build for your interpreter already.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by peterhoeg** [comment]
Closing in favour of #64542.

**Pull request closed without merging**